httpbakery/form/form.go

```diff
@@ -51,49 +53,12 @@ client's responsibility to interpret the schema and present it to the
 user.
 */
 
-// SetUpAuth configures form authentication on c. The VisitWebPage field
-// in c will be set to a function that will attempt form-based
-// authentication using f to perform the interaction with the user and
-// fall back to using the current value of VisitWebPage if form-based
-// authentication is not supported.
-func SetUpAuth(c *httpbakery.Client, f form.Filler) {
-	c.VisitWebPage = VisitWebPage(c, f, c.VisitWebPage)
-}
-
-// VisitWebPage creates a function suitable for use with
-// httpbakery.Client.VisitWebPage. The new function downloads the schema
-// from the specified server and calls f.Fill. The map returned by f.Fill
-// should match the schema specified, but this is not verified before
-// sending to the server. Any errors returned by f.Fill or fallback will
-// not have their cause masked.
-//
-// If the new function detects that form login is not supported by the
-// server and fallback is not nil then fallback will be called to perform
-// the visit.
-func VisitWebPage(d httprequest.Doer, f form.Filler, fallback func(u *url.URL) error) func(u *url.URL) error {
-	v := webPageVisitor{
-		client: &httprequest.Client{
-			Doer: d,
-		},
-		filler:   f,
-		fallback: fallback,
-	}
-	return v.visitWebPage
-}
-
-// webPageVisitor contains the state required by visitWebPage.
-type webPageVisitor struct {
-	client   *httprequest.Client
-	filler   form.Filler
-	fallback func(u *url.URL) error
-}
-
-// loginMethods contains the response expected from the login URL. It
-// only checks for the "form" method as that is the only one that can be
-// handled.
-type loginMethods struct {
-	Form string `json:"form"`
-}
+const (
+	// InteractionMethod is the methodURLs key
+	// used for a URL that can be used for form-based
+	// interaction.
+	InteractionMethod = "form"
```

Comment:
form in this case doesn't mean a form post, but a download of a form schema from the discharger to use to interact with the user. The only known client of this is what it is being modified for, so I don't think backwards compatibilty is a major concern there.